astropy/table/table.py

```diff
@@ -1250,7 +1257,12 @@ def __setitem__(self, item, value):
 
             if isinstance(item, six.string_types):
                 # Set an existing column
-                self.columns[item][:] = value
+                try:
+                    # See definition of mask property for discussion of next line.
+                    assert not getattr(self, '_setitem_inplace', False)
```

Comment:
Ouch, this is good to know. There are a few places (~50) in the code where an assert is used, so you suggest that ideally we should get rid of those?